pkg/storage/chunk/fetcher/fetcher.go

```diff
@@ -230,6 +231,10 @@ func (c *Fetcher) WriteBackCache(ctx context.Context, chunks []chunk.Chunk) erro
 	keysL2 := make([]string, 0, len(chunks))
 	bufsL2 := make([][]byte, 0, len(chunks))
 	for i := range chunks {
+		if c.skipQueryWritebackCacheOlderThan > 0 && chunks[i].From.Time().Before(time.Now().UTC().Add(-c.skipQueryWritebackCacheOlderThan)) {
```

Comment:
nit: return early if the resulting `keys` or `keysL2` is empty

Reply:
Where would we return early? There is already a check if `keysL2` is greater than 0. I can add that to the main cache, but an empty slice passed to the store is a no-op